server/src/main/java/com/cloud/server/ManagementServerImpl.java

```diff
@@ -1442,7 +1442,11 @@ private boolean hasSuitablePoolsForVolume(final VolumeVO volume, final Host host
         StoragePool srcVolumePool = _poolDao.findById(volume.getPoolId());
         allPools = getAllStoragePoolCompatileWithVolumeSourceStoragePool(srcVolumePool);
         allPools.remove(srcVolumePool);
-        suitablePools = findAllSuitableStoragePoolsForVm(volume, vm, srcVolumePool);
+        if (vm != null) {
```

Comment:
One liner improvement if you have some time: `suitablePools = vm == null ? allPools : findAllSuitableStoragePools(....)`

Reply:
I can consider it in the future, I generally like to avoid ternary operator which reduces code readability.